pdns/dnsdist.cc

```diff
@@ -487,7 +493,7 @@ bool processResponse(char** response, uint16_t* responseLen, size_t* responseSiz
   }
 #endif /* HAVE_DNSCRYPT */
 
-  return true;
+  return sendReponse;
```

Comment:
What is the purpose of this change?

Reply:
So this is pretty weird in my opinion. And i guess i missed it on my first explanation because it is kind of complicated.
This function determines if we send the backend reply back to the user. This is weird because the original implementation just sent back every reply from the backend because no messages were allowed to pass to backend if there was cached data. But now we do pass in that case, so I added the check on line 484. But I still need to send a reponse the first time when there is no cached data. so this is why the default is true.
I hope this explains it well.